### DISCUSSION TITLE
Setting a limit on number of assigned values from a domain variable

### DISCUSSION THREAD

**Discussion by ghost** [google/or-tools#2651] [CP-SAT questions] [answered]
I'm trying to use OR-Tools and the CP-SAT solver to solve a problem similar to Sudoku, but instead of limiting the amount of assignments for each number [1,9], each value needs to be assigned a specific (>=1) number of times.

For simplicity's sake, let's say I'm trying to organize lunches for a field trip of 30 students and lost the roster. I have 14 ham sandwiches, 6 tuna sandwiches, and 10 pb&j's. I have information about students' preferences and matching items (e.g. 'no one who got a ham sandwich got a capri sun' or 'student #2 has a peanut allergy') that will help me assign the correct sandwich to each student. However, I can't assign more than the number of sandwiches that I have and I can't assign fewer.

When I try to do:

model.Add(sandwich_column.count('pb') == 10) 
_I've assigned every sandwich a representative integer_

I receive the error message "NotImplementedError: Evaluating a BoundedLinearExpr as a Boolean value is not supported." I understand why this is happening, but I'm not sure of an alternate way of setting this constraint or if it's even possible. I can't use count() on my model at all.

Thanks in advance for your help!

**Comment by lperron** [comment, marked as answer]
Only use Boolean variables, 1 per (var, value) in your original model.

**Reply by ghost** [reply]
Hi Laurent, thank you for your help!

Do you mean something like this?

Student ID1: ham (true/false), tuna (true/false), pb&j (true/false)
-> only one can be true
-> only so many "true" pb&j values for all students

**Reply by lperron** [reply]
yes.

**Reply by ghost** [reply]
Thanks, that makes sense to me. I'll give it a try.
